### DISCUSSION TITLE
[New Feature] Firing delay

### DISCUSSION THREAD

**Discussion by Rampastring** [Vinifera-Developers/Vinifera#1432] [Feature Requests]
### Description:
The engine has no easy way to delay weapons from firing. All vehicles fire right away when a target has been acquired (or given) and the target is in range. Making it possible to delay the firing of units would increase possible balancing options.

For example, instead of having a unit fire instantly as you tell it to fire at an object within range, you could specify the firing to be delayed by X frames after the target has been acquired.

This would be separate from the `ROF` value of the unit's weapon. ROF specifies the delay for the next shot _after_ the unit has fired, while this new functionality would allow specifying the delay _before_ the unit has fired.

### Possible Implementation:
I propose adding a `FiringDelay=` key to `TechnoTypeClass`. The value would be an integer, specifying how many frames the unit needs to wait between acquiring its target and before firing its weapon. 

### Additional Files:
TS already allows delaying units from firing with the `FiringSyncFrame` logic, but that comes with side-effects and some severe drawbacks. @Bittah knows more about it if it's relevant to the issue.

**Comment by Crimsonum** [comment]
I would also like to have associated animation and sound effect support (the latter can be done through the animation if needed).
E.g. `ChargeAnim=<AnimType>` and `ChargeSound=<SoundType>`

For reference, `Charges` allows delayed fire on certain types of units and buildings, but has several caveats:
- It's only applicable to buildings and vehicles with shape graphics.
- It's incompatible with turrets.
- The weapon will not fire until the entire firing animation has played.
